src/components/main/friends/friend/mod.rs

```diff
@@ -27,7 +27,6 @@ pub struct Props<'a> {
 #[allow(non_snake_case)]
 pub fn Friend<'a>(cx: Scope<'a, Props>) -> Element<'a> {
     log::debug!("rendering Friend");
-    let state = use_atom_ref(&cx, STATE);
```

Comment:
Instead of removing it and using it in 2 locations you could simply revert this and clone the state by doing `let stat2 = state.clone()`, etc., and use `state2` in the section location.

Reply:
Done! I also made `remove_friend` function to reduce redundancy.